hooks/boot/31-upgrade-puppet.rb

```diff
@@ -0,0 +1,6 @@
+app_option(
+  '--upgrade-puppet',
```

Comment:
This should contain the target version number (4), so as to leave room for 5 and future versions.